odoo/addons/base/views/res_users_views.xml

```diff
@@ -187,7 +187,7 @@
                             <page name="calendar" string="Calendar">
                                 <group>
                                     <group name="calendar_preferences">
-                                        <field name="tz" widget="timezone_mismatch" options="{'tz_offset_field': 'tz_offset'}" readonly="0" invisible="share"/>
+                                        <field name="tz" widget="timezone_mismatch" options="{'tz_offset_field': 'tz_offset'}" readonly="0" invisible="share" required="1"/>
```

Comment:
For master: Instead of adding it in the view, you should add required on the field.
However, if you just install base, you can set `tz = False` on users.